lib/mix/lib/mix/tasks/compile.ex

```diff
@@ -98,6 +98,26 @@ defmodule Mix.Tasks.Compile do
     |> maybe_prepend(:yecc)
   end
 
+  @doc """
+  Reenables given compilers so they can be executed again down the stack.
+
+  If an umbrella project reenables compilers, they are re-enabled for all
+  child projects.
+
+  By default reenables all compilers.
+  """
+  @spec reenable([compilers: compilers]) :: :ok when compilers: :all | [Mix.Task.task_name()]
```

Comment:
Perhaps better to keep it as an atom. After all, it is not the task name, just part of it.

```suggestion
  @spec reenable([compilers: compilers]) :: :ok when compilers: :all | [atom()]
```